src/rgw/rgw_common.h

```diff
@@ -527,6 +527,14 @@ void encode_json(const char *name, const RGWUserCaps& val, Formatter *f);
 
 void decode_json_obj(obj_version& v, JSONObj *obj);
 
+enum RGWUserType
+{
+  TYPE_NONE=0,
+  TYPE_KEYSTONE,
```

Comment:
@pritha-srivastava I'd put rgw first, then keystone and then ldap. Also, would make the value explicit (even though probably redundant)